api/controllers/WebActionsController.js

```diff
@@ -127,6 +128,83 @@ module.exports = {
     }
   },
 
+  joinSplashList: function(req, res) {
+    let { EMAIL, FNAME, PHONE, group } = req.body;
+    let memberHash = crypto
+      .createHash('md5')
+      .update(EMAIL.toLowerCase())
+      .digest('hex');
+    let redirectUrl = '/confirmation';
+
+    const mailchimpUpdateRequest = {
+      method: 'PATCH',
+      uri: `${sails.config.globals.mailchimpApiUrl}/lists/${sails.config.globals
+        .mailchimpListId}/members/${memberHash}`,
+      json: true,
+      auth: {
+        user: sails.config.globals.mailchimpApiAuthUser,
+        pass: sails.config.globals.mailchimpApiAuthPass,
+      },
+      body: {
+        status: 'subscribed',
+        // Subscribe users to Shine Splash App Group
+        interests: { e8db15c44a: true },
+      },
+    };
+
+    const mailchimpSubscribeRequest = {
+      method: 'POST',
+      uri: `${sails.config.globals.mailchimpApiUrl}/lists/${sails.config.globals
+        .mailchimpListId}/members`,
+      json: true,
+      auth: {
+        user: sails.config.globals.mailchimpApiAuthUser,
+        pass: sails.config.globals.mailchimpApiAuthPass,
+      },
+      body: {
+        email_address: EMAIL,
+        status: 'subscribed',
+        merge_fields: {
+          FNAME: FNAME,
+          PHONE: PHONE,
+        },
+        interests: { e8db15c44a: true }, // Subscribe users to Shine Splash App Group
+      },
+    };
+
+    request
+      .patchAsync(mailchimpUpdateRequest)
+      .then(response => {
+        if (!response || !response.body) {
+          sails.log.error(
+            'Invalid response received from MailChimp subscribe call.'
+          );
+        } else if (response.body.status === 404) {
+          return request.postAsync(mailchimpSubscribeRequest);
+        } else if (response.body.status === 200) {
+          sails.log.info('Successful MailChimp update');
+          res.redirect(redirectUrl);
+        } else {
+          sails.log.error(
+            'Invalid response received from MailChimp subscribe call.'
+          );
+        }
+      })
+      .then(postResonse => {
+        if (!postResonse || !postResonse.body) {
```

Comment:
A `!postResponse` is not necessarily an error case here. The code in the previous `then` I think will still return a Promise regardless but pass on an `undefined` argument.